### PR TITLE
Document handling of optional dependencies

### PR DESCRIPTION
**Pull request by nipafx**
Commit message comes later...

---
**PR checklist**

The following checklist shall help the PR's author, the reviewers and maintainers to ensure the quality of this project.
It is based on our contributors guidelines, especially the ["writing code" section](https://github.com/junit-pioneer/junit-pioneer/blob/main/CONTRIBUTING.md#writing-code).
It shall help to check for completion of the listed points.
If a point does not apply to the given PR's changes, the corresponding entry can be simply marked as done. 

Documentation (general)
* [x] There is documentation (Javadoc and site documentation; added or updated)
* [x] Only one sentence per line (especially in `.adoc` files)
* [x] Javadoc uses formal style, while sites documentation may use informal style

Code
* [x] There are checks which validate correct / false usage / configuration of a functionality and there are tests to verify those checks

Contributing
* [ ] A prepared commit message exists

---

I hereby agree to the terms of the [JUnit Pioneer Contributor License Agreement](https://github.com/junit-pioneer/junit-pioneer/blob/main/CONTRIBUTING.md#junit-pioneer-contributor-license-agreement).

### CODE REVIEW

File: CONTRIBUTING.md
Comment (beatngu13) [inline CONTRIBUTING.md:385]:
```suggestion
* to _integrate_ with other tools, frameworks, and libraries by offering features that directly interact with them (a hypothetical example is [Playwright](https://playwright.dev) for E2E testing)
* for _ease of use_ when recreating functionality would be too complex or otherwise out of scope for Pioneer (a hypothetical example is [Jackson](https://github.com/FasterXML/jackson) for JSON parsing)
```

File: .infra/checkstyle/import-control.xml
Comment (beatngu13) [inline .infra/checkstyle/import-control.xml:6]: Had a quick look at the [docs](https://checkstyle.sourceforge.io/config_imports.html#ImportControl) (on SourceForge 🤭), do I get this right: within `org.junitpionner` (and its "subpackages"), allow everything (`.*`)? (As your commit message says, preparation...)

Comment (Michael1993) [reply .infra/checkstyle/import-control.xml:6]: In the current state of the project we don't _actually_ have optional dependencies. Once this PR gets merged, we can ask contributors (and maintainers?) that:
 - Features using the optional dependency be in their own package (?)
 - Add a line allowing packages from the optional dependency only in the feature-related package (?)
Did I get this right?

**Comment by Michael1993** [comment]
Proposed commit message:
```
Document optional dependency policy #576 / #580

Update CONTRIBUTING.md with the new policy regarding optional dependencies in Pioneer.

Closes: #502
PR: #593
```

File: .infra/checkstyle/import-control.xml
Comment (nipafx) [reply .infra/checkstyle/import-control.xml:6]: Yeah, this is a neutral change that has no practical consequences, but foreshadows future additions. I 100% stole this from #492 because I'd prefer a git blame to point to this commit/PR/issue rather than the "random" extension that first needed it.

**Comment by nipafx** [comment]
Just a heads-up: I want to include much of the exploration done in #502 into the commit message, so please don't merge this yet.

**Review by beatngu13 — APPROVED** [review]

**Pull request merged**
